indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTask.java

```diff
@@ -283,13 +282,20 @@ public boolean withinMinMaxRecordTime(final InputRow row)
   @VisibleForTesting
   public Appenderator getAppenderator()
   {
-    return runner.getAppenderator();
+    return getRunner().getAppenderator();
   }
 
   @VisibleForTesting
   public SeekableStreamIndexTaskRunner<PartitionIdType, SequenceOffsetType> getRunner()
   {
+    if (runner == null) {
+      synchronized (runnerInitLock) {
+        if (runner == null) {
+          runner = createTaskRunner();
+        }
+      }
+    }
```

Comment:
and here I thought that with all java8+ fanciness we would have a method in `AtomicReference<T>.getOrCreate(Supplier<T>)` but it seems this is still the way for lazy singleton.

Reply:
guava has `Suppliers.memoize` that does this, we use it in a couple of places already

Reply:
that sounds nice, now I remember seeing that :)

Reply:
https://github.com/apache/incubator-druid/pull/7740